src/newlines.cpp

```diff
@@ -4936,6 +4953,16 @@ void do_blank_lines(void)
          }
       }
 
+      // Add blanks after namespace
```

Comment:
BTW, is it possible to combine this with the previous change? (They both appear to have the same initial `if` condition?

Reply:
not quite

previous condition is true if current chunk is either `;` or `}` and its parent is `struct`, `union`, `enum` or a `class`; this condition includes `}` only. Of cause, for namespaces there'll be no 
`;` with `CT_NAMESPACE` parent, as we discussed above, but I thought its better to state required conditions explicitly rather than to rely on parsing logic which lies somewhere else and can be changed in future (e.g. when another language is added).

If you think otherwise, let me know and I'll join them.

Reply:
The previous condition is:
```c++
if (chunk_is_token(prev, CT_BRACE_CLOSE) && prev->parent_type == CT_NAMESPACE)
```

This condition is:
```c++
if (chunk_is_token(prev, CT_BRACE_CLOSE) && prev->parent_type == CT_NAMESPACE)
```

Those look the same to me:
```c++
if (chunk_is_token(prev, CT_BRACE_CLOSE) && prev->parent_type == CT_NAMESPACE) // 1
if (chunk_is_token(prev, CT_BRACE_CLOSE) && prev->parent_type == CT_NAMESPACE) // 2
```

Do I miss something? (Based on your comment, did you mean for the first to also check for `CT_SEMICOLON`?)

Anyway, it's just a micro-optimization; not critical.

Reply:
Aha, now I get it.

Thought you meant to join condition on adding blanks after class/union/enum/struct and condition on adding blanks after namespace, but you suggested to join condition for adding blanks before and after namespace.

Done; I merged it all in 8284db4 and force pushed to not make clutter out of many clarification commits.

Reply:
Ah, no... sorry for the confusion :slightly_smiling_face:.